flink-formats/flink-orc/src/main/java/org/apache/flink/orc/vector/AbstractOrcColumnVector.java

```diff
@@ -176,20 +176,15 @@ private static DoubleColumnVector createDoubleVector(int batchSize, Object value
 		return dcv;
 	}
 
-	private static TimestampColumnVector createTimestampVector(int batchSize, LocalDateTime value) {
+	private static TimestampColumnVector createTimestampVector(int batchSize, Object value) {
 		TimestampColumnVector lcv = new TimestampColumnVector(batchSize);
 		if (value == null) {
 			lcv.noNulls = false;
 			lcv.isNull[0] = true;
 			lcv.isRepeating = true;
 		} else {
-			long epochDay = value.toLocalDate().toEpochDay();
-			long nanoOfDay = value.toLocalTime().toNanoOfDay();
-
-			long millisecond = epochDay * 24 * 60 * 60 * 1000 + nanoOfDay / 1_000_000;
-			int nanoOfSecond = (int) (nanoOfDay % 1_000_000_000);
-			Timestamp timestamp = new Timestamp(millisecond);
-			timestamp.setNanos(nanoOfSecond);
+			Timestamp timestamp = value instanceof LocalDateTime ?
+				Timestamp.valueOf((LocalDateTime) value) : (Timestamp) value;
```

Comment:
Maybe it doesn't matter much, but I'm curious if we need to deal with both types.

Reply:
`java.sql.Timestamp` is the default format in hive world, but `LocalDateTime` is the default format in flink world.
Whatever, It must be correct that support all.